graphql/src/test/java/com/linecorp/armeria/server/graphql/ExecutionIdGeneratorTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.graphql;
+
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import graphql.ExecutionResult;
+import graphql.GraphQLContext;
+import graphql.execution.AsyncExecutionStrategy;
+import graphql.execution.ExecutionContext;
+import graphql.execution.ExecutionId;
+import graphql.execution.ExecutionStrategyParameters;
+import graphql.schema.DataFetcher;
+
+class ExecutionIdGeneratorTest {
+
+    static CaptureIdStrategy idStrategy = new CaptureIdStrategy();
+
+    @Nullable
+    static GraphQLContext capturedGraphQLContext;
+
+    static class CaptureIdStrategy extends AsyncExecutionStrategy {
+        @Nullable
+        ExecutionId executionId;
```

Comment:
It's not guaranteed which thread will write to/read from `executionId`

```suggestion
        volatile ExecutionId executionId;
```